service/controller/resources/vaultpki/delete.go

```diff
@@ -32,16 +32,20 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 	}
 
 	if vaultPKIStateToDelete.Backend != nil || vaultPKIStateToDelete.CACertificate != "" {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "deleting the Vault PKI in the Vault API")
+		ids := key.PKIIdsForCluster(key.ClusterID(customObject))
 
-		err := r.vaultPKI.DeleteBackend(key.ClusterID(customObject))
-		if err != nil {
-			return microerror.Mask(err)
-		}
+		for _, id := range ids {
+			r.logger.Debugf(ctx, "deleting the Vault PKI %s in the Vault API", id)
+
+			err := r.vaultPKI.DeleteBackend(id)
```

Comment:
Does `DeleteBackend()` return an error if non-existing backend is tried to be deleted? i.e. can this hang here if on first run only one `id` gets deleted and second one errors but then on the next reconciliation the first id doesn't exist anymore and it gets stuck?